include/aws/mqtt/mqtt.h

```diff
@@ -41,6 +41,7 @@ enum aws_mqtt_error {
     AWS_ERROR_MQTT_UNSUPPORTED_PROTOCOL_LEVEL,
     AWS_ERROR_MQTT_INVALID_CREDENTIALS,
     AWS_ERROR_MQTT_INVALID_QOS,
+    AWS_ERROR_MQTT_INVALID_RETURN_CODE,
```

Comment:
We don't need a new error_code.
If the server sends any value besides 0,1,2,128, it's just an AWS_ERROR_MQTT_PROTOCOL_ERROR.
I see that there are already some distinct error codes for different parts of an MQTT packet (ex: AWS_ERROR_MQTT_INVALID_PACKET_TYPE) but there's no need to continue that trend.